tests/TestableIO.System.IO.Abstractions.TestingHelpers.Tests/MockDirectoryTests.cs

```diff
@@ -747,6 +747,30 @@ public async Task MockDirectory_CreateDirectory_ShouldThrowIfIllegalCharacterInP
         await That(createDelegate).Throws<ArgumentException>();
     }
 
+    [Test]
+    [WindowsOnly(WindowsSpecifics.UNCPaths)]
+    public async Task MockDirectory_CreateDirectory_ShouldSupportExtendedLengthPaths()
+    {
+        // Arrange
+        var fileSystem = new MockFileSystem();
+
+        // Act
+        var directoryInfo = fileSystem.Directory.CreateDirectory(XFS.Path(@"\\?\c:\bar"));
+        fileSystem.File.WriteAllText(@"\\?\c:\bar\grok.txt", "hello world\n");
+
+        // Assert
+        await That(fileSystem.Directory.Exists(XFS.Path(@"\\?\c:\bar"))).IsTrue();
+        await That(directoryInfo.FullName).IsEqualTo(@"\\?\c:\bar");
+        await That(fileSystem.File.ReadAllText(@"\\?\c:\bar\grok.txt")).IsEqualTo("hello world\n");
+        var count = 0;
```

Comment:
You can simplify this check with 

```csharp
        await That(fileSystem.Directory.GetFiles(@"\\?\c:\bar")).HasSingle()
             .Which.IsEqualTo(@"\\?\c:\bar\grok.txt");
```

instead of 

```csharp
        var count = 0;
        foreach (var f in fileSystem.Directory.GetFiles(@"\\?\c:\bar"))
        {
            await That(f).IsEqualTo(@"\\?\c:\bar\grok.txt");
            count++;
        }
        await That(count).IsEqualTo(1);
```